### ISSUE TITLE
jsonld wrong handling of default graph

### ISSUE THREAD

**Issue by jason076** [linkeddata/rdflib.js#310]
I am experimenting with the rdflib and the hydra jsonld API.
### myTestcode.ts
```
import * as rdf from "rdflib"

const store: rdf.IndexedFormula = rdf.graph();
const fetcher: rdf.Fetcher = new rdf.Fetcher(store);
const updateMgr: rdf.UpdateManager = new rdf.UpdateManager(store);
const MY_DOMAIN = rdf.Namespace("https://my-domain.de/vocab#");
const docNode = rdf.sym("https://localhost:8443/companions");
const context = "http://www.w3.org/ns/hydra/context.jsonld";

const jsonDoc ={
    "@context": "http://www.w3.org/ns/hydra/context.jsonld",
    "@id": `${docNode.uri}`,
    "title": "A companion container.",
    "description": "Companions can be managed with ldp container.",
    "operation": [
        {
        "@type": "Operation",
        "method": "POST"
        }
    ]
};

const asNquads = `<https://localhost:8443/companions> <http://www.w3.org/ns/hydra/core#description> "This issue can be deleted with an HTTP DELETE request" .
<https://localhost:8443/companions> <http://www.w3.org/ns/hydra/core#operation> _:b0 .
<https://localhost:8443/companions> <http://www.w3.org/ns/hydra/core#title> "An exemplary issue representation" .
_:b0 <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://www.w3.org/ns/hydra/core#Operation> .
_:b0 <http://www.w3.org/ns/hydra/core#method> "DELETE" .
`
rdf.parse(JSON.stringyfi(jsonldDoc), store, docNode.uri, 'application/ld+json', (err, res) => {
      if (err) {
          throw new Error();
      } else {
          rdf.serialize(docNode, store, docNode.uri, 'application/ld+json', (err, res) => {
              if (err) {
                  throw new Error();
              } else {
                  console.log(res);
              }
          });
     }
});
```

If i am not mistaken, all nodes should belong to the default graph. If I parse a turtle document, rdflib uses the _base_-parameter of the parse function as the graph name. But if I parse the jsonld-document rdflib sets nodeobjects as the graph names. In a result I get a rdf dataset with two different graphs instead of one. 
The line of code where the why (graphname) gets set is in indexed-formular.js lines 270-273:
```
if (!why) {
        // system generated
        why = this.fetcher ? this.fetcher.appNode : this.sym(defaultGraphURI);
      }
}
```
Thus the serializing of _docNode_ does only print 3 rdf-triples instead of 5.

If this behavior is not intended, I could write a fix. I think there must just be a check for the default graph be included in _parser.js_. If the default graph is detected, _why_ will be set to _base_.

**Comment by josephguillaume** [comment]
It seems the quoted lines are now in the add function in store.ts https://github.com/linkeddata/rdflib.js/blob/master/src/store.ts#L422

I also expected parse to use the base parameter as the graph name, but the current implementation of jsonldparser simply doesn't provide a graph name at all, which is why the default is used
https://github.com/linkeddata/rdflib.js/blob/master/src/jsonldparser.js#L87

See comment on relevant commit: https://github.com/linkeddata/rdflib.js/commit/16daad7d68d805aa29e8e01bb4a16d77b28b6afe#r41499395

**Comment by timbl** [comment]
In general, Store.add() should not be called with undefined or null `why`.  Yes, for the top level of a JSON-LD file, the `why` should be set  to a NamedNode of the document being parsed.

**Comment by jeff-zucker** [comment]
Adding the document URI into jsonldParser's createStatement will allow the store to be correctly filled.  Doing that along with Tim's patch to support json-ld in the source pane produces this nice result : 
![image](https://user-images.githubusercontent.com/44732708/126009974-a5c7c03e-e461-447a-ad8c-95df3e482965.png)

**Comment by jeff-zucker** [comment]
I believe this is solved in https://github.com/linkeddata/rdflib.js/commit/80e6b66858b3799ec3efd7cbbfa1394c91c11eb5.  I mistakenly merged it into the jsonld branch directly.

**Comment by timbl** [comment]
Looks good

**Comment by bourgeoa** [comment]
resolved #509
